app/Makefile

```diff
@@ -20,93 +20,64 @@ FLAVOR = debug
 ifndef PDIR # {
 GEN_IMAGES= eagle.app.v6.out
 GEN_BINS= eagle.app.v6.bin
-SPECIAL_MKTARGETS=$(APP_MKTARGETS)
+OPT_MKTARGETS      := coap crypto dht http mqtt pcm sjson sqlite3 tsl2561 u8glib ucglib websocket
```

Comment:
Can confirm, linker fails with lots of undefined references to u8g functions when `LUA_USE_MODULES_U8G` is defined.

Reply:
@devsaurus Arnim, re the warning: this has been there since eLua days, so this is a case of no change.  The NodeMCU make only works under the Xtensa toolchain on Linux, so this is a global issue with our build.  Maybe we should do an OS detect and abort on anything else, But this isn't an issue _introduced_ by this patch.

@pjsg, Agreed and this needs fixing, so I'll have a look at the Travis build and work out what is going wrong, but the assumption in the normal make is that you enable a given module by setting the corresponding define in `user_modulea.h`.  This Travis make seems to be overriding this by a backdoor and forcing all modules into a link, despite the settings of these defines.  This is a QA artiffact as such a biuld will never be capable of being downloaded onto a real ESP module.  It isn't a real-life issue but one of the way the Travis build is configured.  I just need to mirror this short circuit in the make and detect a Travis build and similarly force the build of all of the associated subdirs..

Reply:
> Can confirm, linker fails with lots of undefined references to u8g functions when `LUA_USE_MODULES_U8G` is defined.
I'll take a look at this and fix. :)

Reply:
My tweak to the magic works fine but
```
spiffsimg/spiffsimg -f ../bin/0x%x-4mb.img  -S 4mb -U 0x7c2d0 -r ./spiffsimg/spiffs.lst -d
```
is dying with "Not enough space: fatal error"  Why? swapping the **-S 4mb** for ** -c 0x8000** works fine.

Reply:
OK, I've bottomed this.  The underlying issue is that most of these adjunct directories have the same root name as the module, e.g. `coap` uses `coap`, etc. The two u?g modules don't follow this pattern as `u8g` uses `u8glib` and ditto `ucg`.  I just had to add a couple of extra magic lines to handle this second case.